src/common/settings.ts

```diff
@@ -309,3 +311,56 @@ function getPreferredGlobalSetting<T>(
 
   return newSettings?.defaultValue;
 }
+
+/**
+ * Get the settings that were explicitly set by the user that are only relevant
+ * to the native server.
+ */
+export function getUserSetNativeServerSettings(
+  namespace: string,
+  workspace: WorkspaceFolder,
+): string[] {
+  const settings = [
+    "configuration",
+    "configurationPreference",
+    "exclude",
+    "lineLength",
+    "lint.preview",
+    "lint.select",
+    "lint.extendSelect",
+    "lint.ignore",
+    "lint.extendIgnore",
+    "format.preview",
+  ];
+  const config = getConfiguration(namespace, workspace);
+  return settings
+    .filter((s) => isSettingExplicitlySetByUser(config, s))
+    .map((s) => `${namespace}.${s}`);
+}
+
+/**
+ * Get the settings that were explicitly set by the user that are only relevant
+ * to the legacy server.
+ */
+export function getUserSetLegacyServerSettings(
+  namespace: string,
+  workspace: WorkspaceFolder,
+): string[] {
+  const settings = ["ignoreStandardLibrary", "lint.run", "lint.args", "format.args"];
+  const config = getConfiguration(namespace, workspace);
+  return settings
+    .filter((s) => isSettingExplicitlySetByUser(config, s))
+    .map((s) => `${namespace}.${s}`);
+}
+
+/**
+ * Check if a setting was explicitly set by the user.
+ */
+function isSettingExplicitlySetByUser(config: WorkspaceConfiguration, section: string): boolean {
+  const inspect = config.inspect(section);
+  return (
+    inspect?.globalValue !== undefined ||
+    inspect?.workspaceValue !== undefined ||
+    inspect?.workspaceFolderValue !== undefined
+  );
```

Comment:
The recommended way for testing for `undefined` in JS is
```suggestion
    typeof(inspect?.globalValue) !== "undefined" ||
    typeof(inspect?.workspaceValue) !== "undefined" ||
    typeof(inspect?.workspaceFolderValue) !== "undefined"
  );
```

Reply:
Out of curiosity, why is that so? We use the same pattern elsewhere: https://github.com/astral-sh/ruff-vscode/blob/464ff9b2eec0d197ab845daa8fb79cd32a163cfa/src/common/settings.ts#L253-L313

I couldn't find any reference doing a quick Google search which says that this is the recommended way.

Reply:
Hmm, that might actually just be bad habit from my end... Using `!== undefined` (or just `!= null`) is preferred. https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-typeof-undefined.md